modules/slugs/src/slugs-class-slugs.php

```diff
@@ -374,17 +374,17 @@ function filter_request( $query ) {
 
         // -> page
         elseif ( isset( $query['pagename'] ) || isset( $query['page_id'] ) ):
-            $page = wp_cache_get( 'qts_page_request' );
-            if ( ! $page ) {
+            $page = get_transient( 'qtranslate_slugs_matched_page' );
+            if ( $page === false ) {
                 $page = isset( $query['page_id'] ) ? get_post( $query['page_id'] ) : $this->get_page_by_path( $query['pagename'] );
             }
+            delete_transient( 'qtranslate_slugs_matched_page' );
             if ( ! $page ) {
                 return $query;
             }
             $id          = $page->ID;
             $cache_array = array( $page );
-            update_post_caches( $cache_array, 'page' ); // caching query :)
-            wp_cache_delete( 'qts_page_request' );
+            update_post_caches( $cache_array, 'page' );
```

Comment:
Not sure if it's really needed, also because we are not modifying the post objects here. But I don't think it has anything to do with the `caching query :)` comment btw.
This is calling eventually [wp_cache_add_multiple](https://developer.wordpress.org/reference/functions/wp_cache_add_multiple/), so I think is not updating anything most of the time (it will only update if those keys do not exist already).
We may just add a comment to double check later or remove the calls now...